README.md

```diff
@@ -15,6 +15,35 @@ These are relevant documents related to the content in this repository:
 See [CONTRIBUTING.md](CONTRIBUTING.md) for how to go about contributing to
 this array API standard.
 
+
+## Building docs locally
+
+The spec website comprises of multiple Sphinx docs (one for each spec version),
+all of which exist in `spec/` and rely on the modules found in `src/` (most
+notably `array_api_stubs`). To install these modules and the additional
+dependencies of the Sphinx docs, you can use
+
+```sh
+$ pip install -e .[doc]  # ensure you install the dependencies extra "doc"
```

Comment:
Is it necessary to install the stubs? I would expect the respective conf.py files to add the stubs to the PYTHONPATH automatically.

Reply:
Personally I'd rather have a little more complexity in conf.py if it means the build process is easier (i.e., doesn't require installing the package first).

Reply:
Plus adding stuff to sys.path in conf.py is not all that uncommon. The sys.modules hacks are just a necessary evil due to autosummary being very limited in its configurability. 

Aside: I've spend an inordinate amount of time in the past trying to figure out how to make autosummary not show the full module path for functions, see https://github.com/Chilipp/autodocsumm/issues/66 and https://github.com/sympy/sympy/pull/22589. Messing around with sys.modules is an idea I hadn't considered before. If it works, we should stick with it even if it feels hacky, because it's unlikely we'd find any other solution.

Reply:
> Personally I'd rather have a little more complexity in conf.py if it means the build process is easier (i.e., doesn't require installing the package first).

I think actually we don't benefit much from magic here, as users still need to install the doc dependencies. Documenting in the README to use a single command `pip install -e .[doc]` doesn't seem too bad, even if just having to run `pip install -r requirements.txt` is more intuitive and what we're used to.

Having `sys.path.append("../../array_api_stubs/YYYY_MM/"); import YYYY_MM; sys.modules["array_api"] = YYYY_MM` for all `conf.py` files isn't much a problem and I see not uncommon, but it's still something in terms of maintenance burden and environment hacks. Promoting just a single way to work on both the `array_api_stubs` package and the spec documentation feels a bit nicer IMO as it's more true-to-form for a Sphinx doc and doesn't confuse what the stubs actually are.

Reply:
Also I like the modern `pyproject.toml` process of the `src/` folder which can contain multiple modules. Right now that's just `_array_api_conf`, which gets imported to each `conf.py` and saves us from path hacks per `conf.py`.

I'd also like to utilise `src/` for a small Python script which can be run to build redirect pages in the future (unfortunately `sphinx-reredirects` isn't workable for what we need). Even if we don't do that specifically, it's nice to have the flexibility (i.e. the local development instructions won't change if we introduce a module).